### PR TITLE
Use ROW_NUMBER in JdbcAggregateJobQueryDao to improve performance

### PR DESCRIPTION
**Pull request by onobc**
See #5524

### CODE REVIEW

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/config/AggregateDataFlowTaskConfiguration.java [hunk 1]
```diff
@@ -151,17 +152,13 @@ public TaskDeploymentReader taskDeploymentReader(TaskDeploymentRepository reposi
...
+			JobServiceContainer jobServiceContainer, Environment environment) throws Exception {
+		return new JdbcAggregateJobQueryDao(dataSource, schemaService, jobServiceContainer, environment);
 	}
 
 	@Bean
 	public TaskBatchDaoContainer taskBatchDaoContainer(DataSource dataSource, SchemaService schemaService) {
 		return new TaskBatchDaoContainer(dataSource, schemaService);
 	}
-
-	@PostConstruct
```
Comment (onobc) [inline spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/config/AggregateDataFlowTaskConfiguration.java:163]: This is debug code that we should not need any longer.

File: spring-cloud-dataflow-server/src/test/java/org/springframework/cloud/dataflow/server/db/migration/SqlServer2017SmokeTest.java [hunk 2]
```diff
@@ -19,15 +19,21 @@
 import org.testcontainers.containers.MSSQLServerContainer;
 import org.testcontainers.utility.DockerImageName;
 
+import org.springframework.cloud.dataflow.server.db.ContainerSupport;
+import org.springframework.cloud.dataflow.server.db.oracle.OracleContainerSupport;
```
Comment (cppwfs) [inline spring-cloud-dataflow-server/src/test/java/org/springframework/cloud/dataflow/server/db/migration/SqlServer2017SmokeTest.java:23]: Import can be removed.

File: spring-cloud-dataflow-server/src/test/java/org/springframework/cloud/dataflow/server/db/support/DatabaseTypeTests.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
+package org.springframework.cloud.dataflow.server.db.support;
```
Comment (cppwfs) [inline spring-cloud-dataflow-server/src/test/java/org/springframework/cloud/dataflow/server/db/support/DatabaseTypeTests.java:1]: Missing copyright.

File: spring-cloud-dataflow-test/src/main/java/org/springframework/cloud/dataflow/server/db/ContainerSupport.java [hunk 4]
```diff
@@ -0,0 +1,16 @@
+package org.springframework.cloud.dataflow.server.db;
```
Comment (cppwfs) [inline spring-cloud-dataflow-test/src/main/java/org/springframework/cloud/dataflow/server/db/ContainerSupport.java:1]: Missing Copyright

File: spring-cloud-dataflow-test/src/main/java/org/springframework/cloud/dataflow/server/db/DB2_11_5_ContainerSupport.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2023-2024 the original author or authors.
```
Comment (cppwfs) [inline spring-cloud-dataflow-test/src/main/java/org/springframework/cloud/dataflow/server/db/DB2_11_5_ContainerSupport.java:2]: 2023 for the date.

**Review by corneil — APPROVED** [review]
Great work

**Pull request merged**